core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -2712,7 +2713,15 @@ public void createLanguageFunction(Session session, QualifiedObjectName name, La
         CatalogHandle catalogHandle = catalogMetadata.getCatalogHandle();
         ConnectorMetadata metadata = catalogMetadata.getMetadata(session);
 
-        metadata.createLanguageFunction(session.toConnectorSession(catalogHandle), name.asSchemaFunctionName(), function, replace);
+        SchemaFunctionName schemaFunctionName = name.asSchemaFunctionName();
+        if (catalogMetadata.getSecurityManagement() == SYSTEM) {
+            systemSecurityMetadata.setFunctionOwner(
+                    session,
+                    new CatalogSchemaFunctionName(catalogHandle.getCatalogName().toString(), schemaFunctionName),
+                    new TrinoPrincipal(PrincipalType.USER, session.getUser()));
+        }
+
+        metadata.createLanguageFunction(session.toConnectorSession(catalogHandle), schemaFunctionName, function, replace);
```

Comment:
With views it's done differently - there's a `tableCreated` callback function, which is (presumably) used to update the ownership. The `setXOwner` methods are called in response to `ALTER X SET AUTHORIZATION` commands, instead (which we don't have here). Note that the actual function is created by the connector metadata, which is a factor too.